File: modelos/smart.py
class Smart():

    NAO_INICIADA = 1
    INICIADA = 2
    CONCLUIDA = 3

    STATUS = {
        NAO_INICIADA: 'Não Iniciada',
        INICIADA: 'Iniciada',
        CONCLUIDA: 'Concluída',
    }

    def __init__(self, especifica="", mensuravel="", atingivel="", relevante="", temporizavel="", status=NAO_INICIADA):
        self._especifica = especifica
        self._mensuravel = mensuravel
        self._atingivel = atingivel
        self._relevante = relevante
        self._temporizavel = temporizavel
        self._status = status

    def __str__(self):
        return self._status

File: documentacoes/mongodb/smart.js
print('Executando smart.js');
db = db.getSiblingDB('SMARTify');

db.createCollection("smart", {
    validator: {
        $jsonSchema: {
            bsonType: "object",
            required: ["_id", "usuario_id", "especifica", "mensuravel", "atingivel", "relevante", "temporizavel", "status", "criado_em", "atualizado_em", "deletado"],
            properties: {
                _id:{
                    bsonType: "objectId"
                },
                usuario_id: {
                    bsonType: "string",
                    description: ""
                },
                especifica: {
                    bsonType: ["string", "null"],
                    description: ""
                },
                mensuravel: {
                    bsonType: ["string", "null"],
                    description: ""
                },
                atingivel: {
                    bsonType: ["string", "null"],
                    description: ""
                },
                relevante: {
                    bsonType: ["string", "null"],
                    description: ""
                },
                temporizavel: {
                    bsonType: ["string", "null"],
                    description: ""
                },
                status: {
                    bsonType: "string",
                    enum:["Não Iniciada", "Iniciada", "Concluída"],
                    description: ""
                },
                criado_em: {
                    bsonType: "date",
                    description: ""
                },
                atualizado_em: {
                    bsonType: "date",
                    description: ""
                },
                deletado: {
                    bsonType: "bool",
                    description: ""
                }
            }
        }
    }
});